### PR TITLE
Add data source to retrieve environment variables

### PR DESCRIPTION
**Pull request by LaurentLesle**
## PR Checklist

---

- [ ] I have read the [CONTRIBUTING.MD instructions](./CONTRIBUTING.md)
- [ ] I have changed the `resourceDefinition.json`
- [ ] I have generated the resource model (there's a ```models_generated.go``` file in my PR)
- [ ] I have updated the [README.md#resource-status](../README.md)
- [ ] I have checked to ensure there aren't other open Pull Requests for the same update/change?

## Description

Add support to retrieve an environment variable

```hcl
data "azurecaf_environment_variable" "PAT" {
    name           = "PAT_TOKEN"
    fails_if_empty = true
    sensitive      = true
}
```

## Does this introduce a breaking change

- [ ] YES
- [x] NO

### CODE REVIEW

File: docs/data-sources/azurecaf_environment_variable.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    sensitive      = true
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* name - (required) Name of the environment variable.
+* fails_if_empty (optional) - Through an error if the environment variable is not set (default: false).
```
Comment (arnaudlh) [inline docs/data-sources/azurecaf_environment_variable.md:26]:
```suggestion
* fails_if_empty (optional) - Throws an error if the environment variable is not set (default: false).
```

File: docs/data-sources/azurecaf_environment_variable.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+## Exemple usage
+This example shows how to get the value of an environment variable.
+
+```hcl
+# Retrieve the PATH variable
+data "azurecaf_environment_variable" "path" {
+  name = "PATH"
+}
+
+# Retreive the PAT_TOKEN variable as a sensitive data and through an error if it does not exist.
```
Comment (arnaudlh) [inline docs/data-sources/azurecaf_environment_variable.md:14]:
```suggestion
# Retrieve the PAT_TOKEN variable as a sensitive data and through an error if it does not exist.
```

File: azurecaf/data_environment_variable.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the environment variable.",
+			},
+			"fails_if_empty": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Through an error if the environment variable is not set (default: false).",
```
Comment (arnaudlh) [inline azurecaf/data_environment_variable.go:23]:
```suggestion
				Description: "Throws an error if the environment variable is not set (default: false).",
```

**Review by arnaudlh — CHANGES_REQUESTED** [review]

**Review by arnaudlh — APPROVED** [review]
LGTM

**Comment by BzSpi** [comment]
Do you have any example of use case for this feature ? I'm curious :)

**Pull request merged**